docs/4.2/features/ssh_pam.md

```diff
@@ -0,0 +1,321 @@
+permitted by applicable law.
+```
+
+I've updated this to provide a message to users of Teleport, so they know they are 
+being audited. 
+
+```bash
+$ nano /etc/motd
+WARNING: All activity on this node is being recorded by Teleport
+```
+
+![Teleport SSH with updated MOTD](../img/motd/teleport-with-updated-MOTD.png)
+
+
+## Creating local users to Teleport
+
+Teleport 4.3 introduced the ability to create local (UNIX) users on login. This is
+very helpful if you're a large organization and want to create local user directors on
+the fly. 
+
+Teleport added ability to read in PAM environment variables from PAM handle and pass environment variables to PAM module `TELEPORT_USERNAME`, `TELEPORT_LOGIN`, and `TELEPORT_ROLES`.
+
+This PAM module to creates the user and home directory before attempting to launch 
+a shell for said user.
+
+### Examples
+**Using pam_exec.so**
+
+Using `pam_exec.so` is the easiest way to use the PAM stack to create a user if the
+user does not already exist. The advantage of using `pam_exec.so` is that it usually
+ships with the operating system. The downside is that it doesn't provide access to
+some additional environment variables that Teleport sets (see the `pam_script.so`
+example for those) to use additional identity metadata in the user creation process.
+
+You can either add `pam_exec.so` to your existing PAM stack for your application or 
+write a new one for Teleport. In this example we'll write a new one to simplify how
+to use `pam_exec.so` with Teleport.
+
+Start by creating a file `/etc/pam.d/teleport` with the following contents.
+
+```bash
+account   required   pam_exec.so /etc/pam-exec.d/teleport_acct
+session   required   pam_motd.so
+```
+
+Note the inclusion of `pam_motd.so` under the session facility. While `pam_motd.so` is
+not required for user creation, Teleport requires a module set for both the account
+and session facility to work.
+
+Next create the script that will be run by `pam_exec.so` like below. This script will
+check if the user passed in `PAM_USER` exists and if it does not, it will create it.
+Any error from useradd will be written to `/tmp/pam.error`.
+
+```bash
+mkdir -p /etc/pam-exec.d
+cat > /etc/pam-exec.d/teleport_acct <<EOF
+#!/bin/sh
+id -u "${PAM_USER}" &>/dev/null  || /sbin/useradd -m "${PAM_USER}" 2> /tmp/pam.error
+exit 0
+EOF
+chmod +x /etc/pam-exec.d/teleport_acct
+```
+
+Next update `/etc/teleport.yaml` to call the above PAM stack by both enabling PAM 
+and setting the `service_name`.
+
+```yaml
+ssh_service:
+   pam:
+     enabled: yes
+     service_name: "teleport"
+```
+
+Now attempting to login as an existing user should result in the creation of the user 
+and successful login.
+
+### Using `pam_script.so`
+
+If more advanced functionality is needed `pam_script.so` is a good choice. It typically 
+has to be installed from packages but richer scripts with more identity information 
+from Teleport can be used during the process of user creation.
```

Comment:
If more advanced functionality is needed, `pam_script.so` is a good choice. It typically has to be installed from packages, but enables the use of richer scripts with more identity information from Teleport to be used during the process of user creation.